File: Data_Struct/static/Stack/Stack.java
public class Stack{
	protected Integer size;
	protected Integer last;
	protected char[] vet;

	Stack(Integer size){
		this.size = size;
		vet = new char[size];
		last = -1;
	}

	public boolean empty(){
		if(last == -1) return true;
		return false;
	}

	public Integer size(){
		return last + 1;
	}

	public void push(char elem){
		if(last < size - 1){	
			vet[last+1] = elem;
			last++;
		}else{
			System.out.println("Está cheio");
		}
	}

	public void pop(){
		if(last >= 0){
			last--;
		}else{
			System.out.println("Está vazio");
		}
	}

	public void info(){
		
		for(int i = 0; i <= last; i++){
			System.out.print(i + " ");
		}
		System.out.println();

		for(int i = 0; i <= last; i++){
			System.out.print(vet[i] + " ");	
		}
		System.out.println();
	}

	public char top(){
		if(last != -1){
			return vet[last];
		}
		return '\0';
	}
}

File: Data_Struct/static/Stack/Stack.c
#include <stdio.h>
#include <stdlib.h>

typedef struct{
	int tam;
	int* vet;
	int max;
}Stack;

Stack* newStack(int max){
	Stack* ptr = (Stack*) malloc(sizeof(Stack));
	ptr->tam = 0;
	ptr->max = max;
	ptr->vet = (int*) malloc(max * sizeof(int));
	return ptr;
}

void delStack(Stack* s){
	free(s);
}

void push(Stack* s, int elem){
	int ultimo = s->tam;
	s->vet[ultimo] = elem;
	s->tam = ultimo + 1;
}

void pop(Stack* s){
	s->tam = s->tam - 1;
}

void info(Stack* s){
	int tam = s->tam;
	for(int i = 0; i < tam; i++){
		printf("%d ",s->vet[i]);
	}

	printf("\n");
}

int main(){
	Stack* stack = newStack(15);

	for(int i = 0; i < 15; i++){
		push(stack,i);
	}

	info(stack);

	for(int i = 0; i < 14; i++){
		pop(stack);
		info(stack);
	}
}

